sshkey-attest/src/lib.rs

```diff
@@ -0,0 +1,453 @@
+            AuthenticatorData::try_from(auth_data_bytes.as_slice()).map_err(|e| {
+                error!(?e, "invalid auth data structure");
+                WebauthnError::ParseNOMFailure
+            })?;
+
+        Ok(SshSkAttestation {
+            att_cert,
+            sig,
+            // Probably need auth_data raw.
+            auth_data_bytes,
+            auth_data,
+        })
+    }
+}
+
+fn ssh_sk_attestation_parser(i: &[u8]) -> nom::IResult<&[u8], SshSkAttestationRaw> {
+    // Starts with a 4 byte u32 for the len of the header.
+
+    let (i, _tag_len) = tag([0, 0, 0, 17])(i)?;
+    let (i, _tag) = tag("ssh-sk-attest-v01")(i)?;
+
+    let (i, att_cert_len) = be_u32(i)?;
+    let (i, att_cert_raw) = take(att_cert_len as usize)(i)?;
+
+    let (i, sig_len) = be_u32(i)?;
+    let (i, sig_raw) = take(sig_len as usize)(i)?;
+
+    let (i, auth_data_len) = be_u32(i)?;
+    let (i, auth_data_raw) = take(auth_data_len as usize)(i)?;
+
+    let (i, _resvd_flags) = be_u32(i)?;
+    let (i, _resvd) = be_u32(i)?;
+
+    Ok((
+        i,
+        SshSkAttestationRaw {
+            att_cert_raw,
+            sig_raw,
+            auth_data_raw,
+        },
+    ))
+}
+
+fn to_ssh_pubkey(cose: COSEKey) -> Result<PublicKey, WebauthnError> {
+    match &cose.key {
+        COSEKeyType::EC_EC2(_ec2k) => {
+            let pubkey = cose.get_openssl_pkey()?;
+            let key = pubkey
+                .ec_key()
+                .and_then(|ec| {
+                    let mut ctx = bn::BigNumContext::new()?;
+                    let c_nid = nid::Nid::X9_62_PRIME256V1; // NIST P-256 curve
+                    let group = ec::EcGroup::from_curve_name(c_nid)?;
+
+                    ec.public_key().to_bytes(
+                        &group,
+                        ec::PointConversionForm::UNCOMPRESSED,
+                        &mut ctx,
+                    )
+                })
+                .map_err(WebauthnError::OpenSSLError)?;
+
+            let kind = PublicKeyKind::Ecdsa(EcdsaPublicKey {
+                curve: Curve::from_identifier("nistp256").map_err(|_| {
+                    error!("Invalid curve identifier");
+                    WebauthnError::SshPublicKeyInvalidCurve
+                })?,
+                key,
+                sk_application: Some("ssh:".to_string()),
+            });
+
+            Ok(PublicKey {
+                key_type: KeyType {
+                    name: "sk-ecdsa-sha2-nistp256@openssh.com",
+                    short_name: "ECDSA-SK",
+                    is_cert: false,
+                    is_sk: true,
+                    kind: KeyTypeKind::EcdsaSk,
+                    plain: "sk-ecdsa-sha2-nistp256@openssh.com",
+                },
+                kind,
+                comment: None,
+            })
+        }
+        _ => {
+            debug!("to_ssh_pubkey");
+            Err(WebauthnError::SshPublicKeyInvalidType)
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
Can there be some tests to check for expected attestation failures, seeing as that's security-sensitive?